rfcs/0051-stop-transaction-processing.md

```diff
@@ -0,0 +1,206 @@
+# Stop processing transactions / stop the network after a certain slot
+
+This PR describes the feature to stop processing transactions and to stop the
+network after a certain slot, to be used in the Berkeley hard fork.
+
+## Summary
+
+Transactions come from a client or the gossip network and are processed by BPs
+and SNARK workers to be included in blocks. These blocks are propagated through
+the network and validated by other participants.
+
+In this RFC, the procedure to stop processing any new transactions and to
+stop the network after a certain slot is described. This is, we define two
+slots: the first one is the slot after which any blocks produced will include no
+transaction at all and no fee payments, and the second one is the slot after
+which no blocks are produced and blocks received are rejected.
+
+## Motivation
+
+In a hard fork scenario, we want to halt the preceding network and produce a new
+genesis ledger for the succeeding network. This new genesis ledger should be
+produced from a stabilised staged ledger from the preceding network. This is, we
+define a point in time (slot) where the network continues to operate but with no
+"activity". In detail, after this slot, the network continues to produce blocks
+but without including any transactions, and ensuring any fees paid are 0. This
+will run for a certain number of slots, after which the network will stop
+producing blocks. This will allow the network to stabilise and produce a new
+genesis ledger from the last ledger produced by the network.
+
+This feature enables part of this procedure, by adding the definition of the
+slots and the mechanisms to stop the node from processing transactions and to
+stop the networks after those slots.
+
+## Detailed design
+
+The procedure to stop processing transactions and producing/validating empty
+blocks after a certain slot will be as follows:
+
+* There will be a configuration parameter set at compile-time that will define
+  the slot at which the node will stop processing transactions.
+* The previous configuration should be overridable at runtime by optional CLI
```

Comment:
We were discussing how to test this in an automated fashion recently, and it did come up that we may need to allow this to be overridable at runtime. But there are still security implications there we want to be careful of (eg if this is exposed and adversary could make a malicious version of the config and distribute it, causing any nodes running the config to fork at a specific slot).

This language can stay here for now though, and we can update this decision when we get the the point of automating a test for this.